documents/system-design-document.md

```diff
@@ -0,0 +1,455 @@
+                description:
+                  type: string
+                  description: Detailed description of the task
+                status:
+                  type: string
+                  description: Current status of the task (e.g., 'Created', 'Committed', 'In Progress', 'Completed')
+                user_id:
+                  type: integer
+                  description: ID of the user who owns the task
+
+GET /api/tasks/{task_id}:
+  summary: Retrieves a specific task by its ID
+  parameters:
+    - name: task_id
+      in: path
+      required: true
+      schema:
+        type: integer
+      description: The unique identifier of the task to retrieve
+  responses:
+    '200':
+      description: Task details successfully retrieved
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              id:
+                type: integer
+                description: The unique identifier of the task
+              name:
+                type: string
+                description: The name of the task
+              description:
+                type: string
+                description: Detailed description of the task
+              status:
+                type: string
+                description: Current status of the task (e.g., 'Created', 'Committed', 'In Progress', 'Completed')
+              user_id:
+                type: integer
+                description: ID of the user who owns the task
+    '404':
+      description: Task not found
+    '400':
+      description: Bad request, possibly due to an invalid task ID
+
+PUT /api/tasks/{task_id}:
+  summary: Updates an existing task by its ID, allowing modifications to the name, status, description, or tags
+  parameters:
+    - name: task_id
+      in: path
+      required: true
+      schema:
+        type: integer
+      description: The unique identifier of the task to update
+  requestBody:
+    content:
+      application/json:
+        schema:
+          type: object
+          properties:
+            name:
+              type: string
+              description: The updated name of the task
+            description:
+              type: string
+              description: The updated description of the task
+            status:
+              type: string
+              description: The updated status of the task (e.g., 'Created', 'Committed', 'In Progress', 'Completed')
+            tags:
+              type: array
+              description: A list of tag IDs to associate with the task (add or remove tags)
+              items:
+                type: integer
+          additionalProperties: false
+  responses:
+    '200':
+      description: Task successfully updated
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              success:
+                type: boolean
+                example: true
```

Comment:
Updated content would be helpful for the client, maybe simply the `Task` object with updated values.